File: src/main/java/net/softbell/bsh/handler/security/CustomAccessDeniedHandler.java
package net.softbell.bsh.handler.security;

import java.io.IOException;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

import org.springframework.security.access.AccessDeniedException;
import org.springframework.security.web.access.AccessDeniedHandler;
import org.springframework.stereotype.Component;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.NoArgsConstructor;
import lombok.extern.slf4j.Slf4j;

@Slf4j
@Component
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CustomAccessDeniedHandler implements AccessDeniedHandler
{
	// Global Field
	private String G_API_URI, G_VIEW_URI;
	
    @Override
    public void handle(HttpServletRequest request, HttpServletResponse response, AccessDeniedException exception)
    		throws IOException
    {
    	// Field
    	boolean isAPI = false;
    	
    	// Init
    	if (request.getRequestURI().startsWith("/api/"))
    		isAPI = true;
    	
    	// Redirect
    	if (isAPI)
    		response.sendRedirect(G_API_URI);
    	else
    		response.sendRedirect(G_VIEW_URI);
    }
}

File: src/main/kotlin/net/softbell/bsh/handler/security/CustomAccessDeniedHandler.kt
package net.softbell.bsh.handler.security

import org.springframework.security.access.AccessDeniedException
import org.springframework.security.web.access.AccessDeniedHandler
import org.springframework.stereotype.Component
import java.io.IOException
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse

/**
 * @author : Bell(bell@softbell.net)
 * @description : 접근 제한 핸들러
 */
@Component
class CustomAccessDeniedHandler(
        private val API_URI: String = "",
        private val VIEW_URI: String = ""
) : AccessDeniedHandler {
    @Throws(IOException::class)
    override fun handle(request: HttpServletRequest, response: HttpServletResponse, exception: AccessDeniedException) {
        // Field
        var isAPI = false

        // Init
        if (request.requestURI.startsWith("/api/"))
            isAPI = true

        // Redirect
        if (isAPI)
            response.sendRedirect(API_URI)
        else
            response.sendRedirect(VIEW_URI)
    }
}